File: gsd09-web-crawler/common/event/src/main/java/event/event/payload/UrlDiscoveryCreatePayload.java
package event.event.payload;

import event.event.EventPayload;
import lombok.Getter;

@Getter
public class UrlDiscoveryCreatePayload implements EventPayload {
    private String crawlingId;
    private Integer depth;
    private Integer maxDepth;
    private String url;
    private String bfsPath;

    public static UrlDiscoveryCreatePayload create(String crawlingId,
                                                   Integer depth,
                                                   Integer maxDepth,
                                                   String url,
                                                   String bfsPath) {
        UrlDiscoveryCreatePayload payload = new UrlDiscoveryCreatePayload();
        payload.crawlingId = crawlingId;
        payload.depth = depth;
        payload.maxDepth = maxDepth;
        payload.url = url;
        payload.bfsPath = bfsPath;
        return payload;
    }
}

File: sdip09-web-crawler-ktor/common/event/src/main/kotlin/event/UrlDiscoveryCreatePayload.kt
package event

import kotlinx.serialization.Serializable

@Serializable
data class UrlDiscoveryCreatePayload(
    val crawlingId: String,
    val depth: Int,
    val maxDepth: Int,
    val url: String,
    val bfsPath: String
) : EventPayload {

    companion object {
        fun create(
            crawlingId: String,
            depth: Int,
            maxDepth: Int,
            url: String,
            bfsPath: String
        ): UrlDiscoveryCreatePayload = UrlDiscoveryCreatePayload(
            crawlingId = crawlingId,
            depth = depth,
            maxDepth = maxDepth,
            url = url,
            bfsPath = bfsPath
        )
    }
}
